src/cspviolationdetector/DefaultCSPViolationDetector.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import CSPViolationDetector from './CSPViolationDetector';
+
+export default class DefaultCSPViolationDetector implements CSPViolationDetector {
```

Comment:
Do we really need an interface/default split here? It's a bad pattern.

Reply:
I follow the pattern from existing code.

Reply:
Not always a good thing to do…

Reply:
Sure, I will check if we can use a better pattern to create and use this component. Thanks for pointing it out.

Reply:
Unfortunately it is restricted in the code to add an interface for a new component like this.

Reply:
Fix the rule, because that's a bad policy.

Here's the problem. Each time we define an interface and then a class that implements it, that interface becomes part of our public, fixed API surface. We can't add to it because a customer might have used the interface! So because we have rules:

1. Each class should go in a separate file.
2. Each class has a matching interface.

Our SDK has an enormous public API surface area, which is bad for customers and it's bad for us.

We don't want customers to define their own `CSPViolationDetector` using our interface, and we do not intend to have more than one implementation, so we should not make the interface public… or have an interface at all.

Fix the rule.

Reply:
As discussed, will fix the rule in a separate pr. For this `CSPViolationDetector` component, I will make an exemption to it so that it doesn't need to have an matching interface.